core_backend/tests/api/conftest.py

```diff
@@ -33,29 +31,44 @@
 
 
 @pytest.fixture(scope="session")
-def faq_contents(client: TestClient) -> None:
+def db_session() -> pytest.FixtureRequest:
+    """Create a test database session."""
+    session_gen = get_session()
+    session = next(session_gen)
+
+    try:
+        yield session
+    finally:
+        session.rollback()
+        next(session_gen, None)
```

Comment:
Oo. Could you explain how this works? get_session() returns a generator? What does the code for finally statement do?

Reply:
Yes get_session returns a generator. I wasn't able to use a context manager to build the fixture. What the code does:
- get the session as a generator,
- Get the first element which is the actual sesson
- Then, yield the session for the whole test
- Then `session.rollback()` to undo the changes made in the database 
- Then we release the session with `next(session_gen, None)`